sdk-core/src/main/java/dev/restate/sdk/core/InvocationStateMachine.java

```diff
@@ -202,7 +203,10 @@ void onStart(MessageLite msg) {
     this.afterStartCallback.consume(cb -> cb.accept(new InvocationIdImpl(this.debugId)));
   }
 
-  void close() {
+  /**
+   * @param suspensionIndexes if null, we're not closing with a suspension
+   */
+  void close(@Nullable Iterable<Integer> suspensionIndexes) {
```

Comment:
Should we have methods `suspend` and `end` to make it more explicit?

Reply:
Yep, applied that.